lib/lunar.js

```diff
@@ -163,36 +168,46 @@ function altitude_julian(h, event, t, lat, lon) {
     }
 
     if (roots === 1) {
-      if (h0 < 0) { 
-        times.rise = i + x1;
-      } else {
-        times.set = i + x1;
+      if (h0 < 0 && isNaN(times.moonrise)) {
+        times.moonrise = i + x1;
+      } else if(isNaN(times.moonset)) {
+        times.moonset = i + x1;
       }
     } else if (roots === 2) {
-      times.rise = i + (ye < 0 ? x2 : x1);
-      times.set  = i + (ye < 0 ? x1 : x2);
+      times.moonrise = isNaN(times.moonrise) ?
+          i + (ye < 0 ? x2 : x1) : times.moonrise;
+      times.moonset  = isNaN(times.moonset) ?
+          i + (ye < 0 ? x1 : x2): times.moonset;
     }
 
-    // Found what we were looking for
-    if (times[event] && times[event] < 24) {
-      return hours_later(t, times[event]);
+    // Found all the things!
+    if(!isNaN(times.moonrise) && !isNaN(times.moonset)) {
+      if(times.moonrise >= 24) {
+        times.moonrise = NaN;
+      } else {
+        times.moonrise = hours_later(t, times.moonrise);
+      }
+      if(times.moonset >= 24) {
+        times.moonrise = NaN;
+      } else {
+        times.moonset = hours_later(t, times.moonset);
+      }
+      return times;
     }
     // Move two hours of altitude
     h0 = h2;
   }
 
-  return NaN;
+  return times;
 }
 
-const PARALLAX =  0.0023212879051524586;
-
 /* In the next 24 hours */ 
 function rise(t, lat, lon) {
-  return altitude_julian(-PARALLAX, RISE, t, lat, lon);
+  return rise_and_set(t, lat, lon)[RISE];
 }
 /* In the next 24 hours */
 function set(t, lat, lon) {
-  return altitude_julian(-PARALLAX, SET, t, lat, lon);
+  return rise_and_set(t, lat, lon)[SET];
```

Comment:
Personally I'd just ditch the `RISE` and `SET` constants entirely!

Otherwise I like how this is wrapped and how it's exposed. I'm not worried about it being tons slower in the case we only want one (for some weird reason).